op-supervisor/supervisor/backend/processors/chain_processor.go

```diff
@@ -285,7 +285,18 @@ func (s *ChainProcessor) process(ctx context.Context, next eth.BlockRef, receipt
 		if err := s.rewinder.Rewind(s.chain, next.ParentID()); err != nil {
 			// If any logs were written, our next attempt to write will fail and we'll retry this rewind.
 			// If no logs were written successfully then the rewind wouldn't have done anything anyway.
-			s.log.Error("Failed to rewind after error processing block", "block", next, "err", err)
+			s.log.Error("Failed to rewind after error processing block", "block", next, "parent", next.ParentID(), "err", err)
+		} else {
+			s.log.Warn("Successfully rewinded database to the previous block", "block", next, "parent", next.ParentID())
+
+			s.log.Warn("Emit an invalidate local-safe event", "block", next, "parent", next.ParentID())
+			s.emitter.Emit(superevents.InvalidateLocalSafeEvent{
+				ChainID: s.chain,
+				Candidate: types.DerivedBlockRefPair{
+					Derived: next,
+					Source:  next,
```

Comment:
This is technically incorrect, but `Source` is used only for logging and should probably just be an optional field... here we don't really know the `Source`.

In any case this should be fixed in this PR.

Reply:
Where possible we'd like to take Block Pairs since they are easier to validate than just seeing one or the other field. But to your point if we are only logging it, I agree we could just use a single BlockRef